### ISSUE TITLE
Multi-level treegrid

### ISSUE THREAD

**Issue by jonasrotilli** [vaadin/flow-components#3147]
### Ignore

ignore

**Comment by mvysny** [comment]
Yes, you can have the parent, child and neto beans implement the same interface which would provide values for tree grid rows, and a list of children.

**Comment by web-padawan** [comment]
Hi, yes it is possible to have as many levels as you need:

<img width="1413" alt="treegrid" src="https://user-images.githubusercontent.com/10589913/167583081-71fdf75e-0380-4c56-bd0c-4cee15857816.png">

Please check out [integration tests](https://github.com/vaadin/flow-components/tree/master/vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/treegrid/it) for some examples.

**Comment by yuriy-fix** [comment]
We should consider adding an example to vaadin.com/docs.

**Comment by jonasrotilli** [comment]
I found, by accident, several examples of using vaadin. Very good, should be more publicized: https://cookbook.vaadin.com/

Specifically about the multi-level TreeGrid it is here:
https://cookbook.vaadin.com/tree-grid-recursive-selection
